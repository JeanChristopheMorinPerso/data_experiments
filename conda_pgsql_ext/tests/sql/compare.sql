CREATE EXTENSION condaversion;
SELECT '1.0.0'::condaversion = '1.0.0'::condaversion;

-- Just to prove that we are not comparing strings. This will be true.
SELECT '1.0'::condaversion = '1.0.0'::condaversion;

SELECT '1.2.3'::condaversion > '4.5'::condaversion;
SELECT '4.5'::condaversion > '4.5'::condaversion;
SELECT '6.7'::condaversion > '4.5'::condaversion;

SELECT '1.2.3'::condaversion >= '4.5'::condaversion;
SELECT '4.5'::condaversion >= '4.5'::condaversion;
SELECT '6.7'::condaversion >= '4.5'::condaversion;

SELECT '1.2.3'::condaversion < '4.5'::condaversion;
SELECT '4.5'::condaversion < '4.5'::condaversion;
SELECT '6.7'::condaversion < '4.5'::condaversion;

SELECT '1.2.3'::condaversion <= '4.5'::condaversion;
SELECT '4.5'::condaversion <= '4.5'::condaversion;
SELECT '6.7'::condaversion <= '4.5'::condaversion;

SELECT '1.2.3'::condaversion != '4.5'::condaversion;
SELECT '4.5'::condaversion != '4.5'::condaversion;
SELECT '6.7'::condaversion != '4.5'::condaversion;

SELECT '1.2.3'::condaversion = '4.5'::condaversion;
SELECT '4.5'::condaversion = '4.5'::condaversion;
SELECT '6.7'::condaversion = '4.5'::condaversion;
