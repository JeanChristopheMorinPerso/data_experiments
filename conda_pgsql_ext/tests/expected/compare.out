CREATE EXTENSION condaversion;
SELECT '1.0.0'::condaversion = '1.0.0'::condaversion;
 ?column? 
----------
 t
(1 row)

-- Just to prove that we are not comparing strings. This will be true.
SELECT '1.0'::condaversion = '1.0.0'::condaversion;
 ?column? 
----------
 t
(1 row)

SELECT '1.2.3'::condaversion > '4.5'::condaversion;
 ?column? 
----------
 f
(1 row)

SELECT '4.5'::condaversion > '4.5'::condaversion;
 ?column? 
----------
 f
(1 row)

SELECT '6.7'::condaversion > '4.5'::condaversion;
 ?column? 
----------
 t
(1 row)

SELECT '1.2.3'::condaversion >= '4.5'::condaversion;
 ?column? 
----------
 f
(1 row)

SELECT '4.5'::condaversion >= '4.5'::condaversion;
 ?column? 
----------
 t
(1 row)

SELECT '6.7'::condaversion >= '4.5'::condaversion;
 ?column? 
----------
 t
(1 row)

SELECT '1.2.3'::condaversion < '4.5'::condaversion;
 ?column? 
----------
 t
(1 row)

SELECT '4.5'::condaversion < '4.5'::condaversion;
 ?column? 
----------
 f
(1 row)

SELECT '6.7'::condaversion < '4.5'::condaversion;
 ?column? 
----------
 f
(1 row)

SELECT '1.2.3'::condaversion <= '4.5'::condaversion;
 ?column? 
----------
 t
(1 row)

SELECT '4.5'::condaversion <= '4.5'::condaversion;
 ?column? 
----------
 t
(1 row)

SELECT '6.7'::condaversion <= '4.5'::condaversion;
 ?column? 
----------
 f
(1 row)

SELECT '1.2.3'::condaversion != '4.5'::condaversion;
 ?column? 
----------
 t
(1 row)

SELECT '4.5'::condaversion != '4.5'::condaversion;
 ?column? 
----------
 f
(1 row)

SELECT '6.7'::condaversion != '4.5'::condaversion;
 ?column? 
----------
 t
(1 row)

SELECT '1.2.3'::condaversion = '4.5'::condaversion;
 ?column? 
----------
 f
(1 row)

SELECT '4.5'::condaversion = '4.5'::condaversion;
 ?column? 
----------
 t
(1 row)

SELECT '6.7'::condaversion = '4.5'::condaversion;
 ?column? 
----------
 f
(1 row)

